src/main.rs

```diff
@@ -128,6 +148,108 @@ fn gen(cfg: &Config, markdown: String, out: impl Write) {
     }
 }
 
+fn ffmpeg<P: AsRef<Path>>(pdf: P, cfg: &Config) -> PathBuf {
+    // Start by demuxing the pdf into its frames, which gives us their number.
+    // We use poppler tools and not imagemagick because the latter is bloody stupid. It
+    // disables code access to pdf conversion as a 'security measure'. That's an idiots patch
+    // to broken code that just hurts usability. There's nothing malicious about what we're
+    // attempting to do but, I assumed, since they can't admit to the PHP crowds of SaaS image
+    // tools that their tools are utterly messy and completely unmaintainable they introduce a
+    // black-list that makes it impossible to use for other people as well.
+    // /rant
+    Command::new("pdftoppm")
+        .current_dir(&cfg.out_dir)
+        .arg("-png")
+        .arg(pdf.as_ref())
+        .arg("pages")
+        .status()
+        .expect("Converting pdf with `convert` failed");
+
+    // Then, generate all voice files and record their lengths. Note that a voice file need not
+    // exist (e.g. for title frames) in which case we should prepare some filler (TODO).
+    // Also add all the pages to readable image list for ffmpeg
+    let mut control = File::create(cfg.out_dir.join("ffmpeg.concat.txt"))
+        .expect("Failed to create ffmpeg control file");
+
+    let mut audios = vec![];
+
+    for idx in 0.. {
+        // Yes, for some reason the page index is 1 based
+        let frame = format!("pages-{}.png", idx + 1);
+        let speak = format!("espeak_{}.txt", idx);
+        let wav = format!("espeak-{}.wav", idx);
+
+        if !cfg.out_dir.join(&frame).exists() {
+            break;
+        }
+
+        if !cfg.out_dir.join(&speak).exists() {
+            writeln!(control, "file '{}'", frame).unwrap();
+            writeln!(control, "duration {}", 0.0).unwrap();
+            continue;
+        }
+
+        Command::new("espeak-ng")
+            .current_dir(&cfg.out_dir)
+            .arg("-f")
+            .arg(&speak)
+            .arg("-w")
+            .arg(&wav)
+            .args(&["-v", "Henrique"])
+            .status()
+            .expect("Conversion with `espeak-ng` failed.");
+
+        let output = Command::new("sox")
```

Comment:
If we depend on `ffmpeg` anyway, why don't we use `ffprobe` / the ffmpeg tool suite to get the wav's metadata?

Reply:
It was non-obvious how to do this with `ffmpeg`, if you have some pointers to the documentation then I'll gladly switch the implementation.

Reply:
While it's possible with `ffmpeg`, using `ffprobe` is probably easier (and `ffprobe` is also part of (at least the arch) ffmpeg installs. You can see both solutions [here](https://superuser.com/a/945604).